src/libraries/System.Runtime/tests/System/TimeZoneInfoTests.cs

```diff
@@ -81,6 +81,47 @@ public static void Names()
             Assert.NotNull(utc.ToString());
         }
 
+        //  Due to ICU size limitations, full daylight/standard names are not included.
+        //  name abbreviations, if available, are used instead
+        public static IEnumerable<object []> Platform_TimeZoneNamesTestData()
+        {
+            if (PlatformDetection.IsBrowser)
+                return new TheoryData<TimeZoneInfo, string, string, string>
+                {
+                    { TimeZoneInfo.FindSystemTimeZoneById(s_strPacific), "(UTC-08:00) PST", "PST", "PDT" },
+                    { TimeZoneInfo.FindSystemTimeZoneById(s_strSydney), "(UTC+10:00) AEST", "AEST", "AEDT" },
+                    { TimeZoneInfo.FindSystemTimeZoneById(s_strPerth), "(UTC+08:00) AWST", "AWST", "AWDT" },
+                    { TimeZoneInfo.FindSystemTimeZoneById(s_strIran), "(UTC+03:30) +0330", "+0330", "+0430" },
+
+                    { s_NewfoundlandTz, "(UTC-03:30) NST", "NST", "NDT" },
+                    { s_catamarcaTz, "(UTC-03:00) -03", "-03", "-02" }
+                };
+            else
+                return new TheoryData<TimeZoneInfo, string, string, string>
+                {
+                    { TimeZoneInfo.FindSystemTimeZoneById(s_strPacific), "(UTC-08:00) Pacific Standard Time", "Pacific Standard Time", "Pacific Daylight Time" },
+                    { TimeZoneInfo.FindSystemTimeZoneById(s_strSydney), "(UTC+10:00) Australian Eastern Standard Time", "Australian Eastern Standard Time", "Australian Eastern Daylight Time" },
+                    { TimeZoneInfo.FindSystemTimeZoneById(s_strPerth), "(UTC+08:00) Australian Western Standard Time", "Australian Western Standard Time", "Australian Western Daylight Time" },
+                    { TimeZoneInfo.FindSystemTimeZoneById(s_strIran), "(UTC+03:30) +0330", "+0330", "+0430" },
```

Comment:
Shouldn't these last three be the full names from ICU?

```csharp
{ TimeZoneInfo.FindSystemTimeZoneById(s_strIran), "(UTC+03:30) Iran Standard Time", "Iran Standard Time", "(UTC+03:30) Iran Standard Time" },
{ s_NewfoundlandTz, "(UTC-03:30) Newfoundland Standard Time", "Newfoundland Standard Time", "(UTC-03:30) Newfoundland Standard Time" },
{ s_catamarcaTz, "(UTC-03:00) Argentina Standard Time", "Argentina Standard Time", "(UTC-03:00) Argentina Standard Time" }
```

(Sorry I didn't see this earlier)

Reply:
How did the test pass with the wrong values?

Reply:
Are you referring to the last value in each test? They're supposed to be the DaylightNames.

Reply:
No, I meant that in the non-browser version, the last three rows were not using the ICU names.  I believe its due to the `if (PlatformDetection.IsBrowser)` in the test - this part of the test isn't being run.  Per our offline discussion, I'll fix that up in my PR.